android/modules/media/src/java/ti/modules/titanium/media/TiCameraXActivity.java

```diff
@@ -0,0 +1,570 @@
+						if (videoMaximumSize > 0) {
+							fileOutputOptions.setFileSizeLimit(videoMaximumSize);
+						}
+						videoCapture = VideoCapture.withOutput(recorder);
+						pendingRecording = videoCapture.getOutput()
+							.prepareRecording(activity, fileOutputOptions.build());
+					}
+
+					camera = cameraProvider.bindToLifecycle(this, cameraSelector, videoCapture, preview);
+				} else {
+					// photo
+					imageCapture =
+						new ImageCapture.Builder().setFlashMode(cameraFlashMode).setTargetRotation(rotation).build();
+
+					camera = cameraProvider.bindToLifecycle(this, cameraSelector, imageCapture, preview);
+				}
+				preview.setSurfaceProvider(viewFinder.getSurfaceProvider());
+
+				if (allowZoom) {
+					// zoom gesture
+					ScaleGestureDetector scaleGestureDetector = new ScaleGestureDetector(activity,
+						new ScaleGestureDetector.SimpleOnScaleGestureListener()
+						{
+							@Override
+							public boolean onScale(ScaleGestureDetector detector)
+							{
+								float scale = camera.getCameraInfo().getZoomState()
+									.getValue().getZoomRatio() * detector.getScaleFactor();
+								camera.getCameraControl().setZoomRatio(scale);
+								return true;
+							}
+						});
+
+					viewFinder.setOnTouchListener((v, event) -> {
+						scaleGestureDetector.onTouchEvent(event);
+						v.performClick();
+						return true;
+					});
+				}
+			} catch (InterruptedException | ExecutionException e) {
+				// Currently no exceptions thrown. cameraProviderFuture.get()
+				// shouldn't block since the listener is being called, so no need to
+				// handle InterruptedException.
+			}
+		}, ContextCompat.getMainExecutor(activity));
+	}
+
+	@Override
+	protected void onResume()
+	{
+		super.onResume();
+		cameraActivity = this;
+		if (localOverlayProxy != null) {
+			View overlayView = localOverlayProxy.getOrCreateView().getNativeView();
+			ViewGroup parent = (ViewGroup) overlayView.getParent();
+			if (parent != null) {
+				parent.removeView(overlayView);
+			}
+			layout.addView(overlayView, new FrameLayout.LayoutParams(FrameLayout.LayoutParams.MATCH_PARENT,
+				FrameLayout.LayoutParams.MATCH_PARENT));
+		}
+	}
+
+	@Override
+	protected void onDestroy()
+	{
+		// Release our camera activity reference.
+		if (cameraActivity == this) {
+			cameraActivity = null;
+		}
+		if (recording != null) {
+			recording.close();
+			recording = null;
+		}
+		if (pendingRecording != null) {
+			pendingRecording = null;
+		}
+		if (camera != null) {
+			camera = null;
+		}
+		// Destroy this activity.
+		super.onDestroy();
+	}
+
+	@Override
+	public void finish()
+	{
+		overlayProxy = null;
+		super.finish();
+	}
+
+	@Override
+	public void onBackPressed()
+	{
+		if (androidbackCallback != null) {
+			KrollDict response = new KrollDict();
+			response.putCodeAndMessage(-1, "User pressed androidback");
+			androidbackCallback.callAsync(callbackContext, response);
+		} else {
+			if (cancelCallback != null) {
```

Comment:
Combine if-statements

Reply:
Think that won't work because the `finish()` is called in the else part called even if `cancelCallback` is null

Reply:
Oh, I missed the `finish` call!